src/System.Memory/tests/ParsersAndFormatters/Parser/TestData.Parser.Date.cs

```diff
@@ -12,7 +13,32 @@ public static IEnumerable<ParserTestData<DateTime>> DateTimeParserTestData
         {
             get
             {
-                foreach (ParserTestData<DateTime> testData in DateTimeFormatterTestData.ToParserTheoryDataCollection())
+				// Format J tests for DateTime parsing.
+				yield return new ParserTestData<DateTime>("0997-07-16", DateTime.Parse("0997-07-16"), 'J', expectedSuccess: true);
+				yield return new ParserTestData<DateTime>("1997-07-16", DateTime.Parse("1997-07-16"), 'J', expectedSuccess: true);
+				yield return new ParserTestData<DateTime>("1997-07-16T19:20", DateTime.Parse("1997-07-16T19:20"), 'J', expectedSuccess: true);
+				yield return new ParserTestData<DateTime>("1997-07-16T19:20:30", DateTime.Parse("1997-07-16T19:20:30"), 'J', expectedSuccess: true);
+				yield return new ParserTestData<DateTime>("1997-07-16T19:20:30.45", DateTime.Parse("1997-07-16T19:20:30.45"), 'J', expectedSuccess: true);
+				yield return new ParserTestData<DateTime>("1997-07-16T19:20:30.4555555", DateTime.Parse("1997-07-16T19:20:30.4555555"), 'J', expectedSuccess: true);
+                
+                // Test fraction rounding.
+				yield return new ParserTestData<DateTime>("1997-07-16T19:20:30.45555554", DateTime.Parse("1997-07-16T19:20:30.45555554"), 'J', expectedSuccess: true);
+                // DateTime.Parse will round up to 7dp in this case, but we expect parser to truncate.
+				yield return new ParserTestData<DateTime>("1997-07-16T19:20:30.45555555", DateTime.Parse("1997-07-16T19:20:30.4555555"), 'J', expectedSuccess: true);
+
+				// Test with timezone designator.
+				yield return new ParserTestData<DateTime>("1997-07-16T19:20:30.4555555Z", DateTime.ParseExact("1997-07-16T19:20:30.4555555Z", "O", CultureInfo.InvariantCulture, DateTimeStyles.RoundtripKind), 'J', expectedSuccess: true);
+				yield return new ParserTestData<DateTime>("1997-07-16T19:20:30.4555555+01:00", DateTime.ParseExact("1997-07-16T19:20:30.4555555+01:00", "O", CultureInfo.InvariantCulture, DateTimeStyles.RoundtripKind), 'J', expectedSuccess: true);
+				yield return new ParserTestData<DateTime>("1997-07-16T19:20:30.4555555-01:00", DateTime.ParseExact("1997-07-16T19:20:30.4555555-01:00", "O", CultureInfo.InvariantCulture, DateTimeStyles.RoundtripKind), 'J', expectedSuccess: true);
```

Comment:
I'd recommend adding tests with non-zero minutes in the timezone, for both negative and positive timezones, e.g. +04:30 and -04:30.

Reply:
And +24:00 is a valid ISO 8061 date. Add a test for that.

Reply:
> Midnight is a special case and may be referred to as either "00:00" or "24:00". The notation "00:00" is used at the beginning of a calendar day and is the more frequently used. At the end of a day use "24:00". "2007-04-05T24:00" is the same instant as "2007-04-06T00:00" (see Combined date and time representations below).

https://en.wikipedia.org/wiki/ISO_8601

Reply:
Opps, it isn't the timezone that supports +24:00, it is the hour.

Reply:
You should have tests for all of these timezone situations:

```
<time>Z
<time>±hh:mm
<time>±hhmm
<time>±hh
```

Reply:
> You should have tests for all of these timezone scenarios:

Per the standard, the timezone designator is one of 

> `Z`or `+hh:mm `or `-hh:mm`

https://www.w3.org/TR/NOTE-datetime

We have tested support for those.

Reply:
That spec is 20 years old. It has been updated since then - https://en.wikipedia.org/wiki/ISO_8601#Time_offsets_from_UTC

Reply:
Filed an issue for this https://github.com/dotnet/corefx/issues/35845.